src/Sema.zig

```diff
@@ -32160,6 +32160,30 @@ fn analyzeSlice(
             if (!end_is_len) {
                 const end = if (by_length) end: {
                     const len = try sema.coerce(block, Type.usize, uncasted_end_opt, end_src);
+                    if (try sema.resolveValue(len)) |slice_len_val| {
+                        const len_s_val = try mod.intValue(
+                            Type.usize,
+                            array_ty.arrayLenIncludingSentinel(mod),
+                        );
+                        if (!(try sema.compareScalar(slice_len_val, .lte, len_s_val, Type.usize))) {
+                            const sentinel_label: []const u8 = if (array_ty.sentinel(mod) != null)
+                                " +1 (sentinel)"
+                            else
+                                "";
+
+                            return sema.fail(
+                                block,
+                                end_src,
+                                "length {}{s} array cannot be sliced to length {}",
```

Comment:
Could this error message be made more similar to the related `"end index {} out of bounds for array of length {}{s}"` errors?

Reply:
I presume you're thinking that it would be better to have the order of array length and slice operation length the same as other messages? I'll have more of a think about it - I wasn't totally happy with what I came up with, but I haven't yet found a wording that was clear whilst being similar without being long.

Reply:
I've changed the message to `"length {} out of bounds for array of length {}{s}"` - I think that should be a better match while communicating it is a by-length slice.